### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by raihan2bd**
Hi there 👋 ,
In this branch, I made the following changes ⬇️ 
- Created a new class called `Classroom` and added relationship between `Student` and `Classroom` according to the project requirement ✅ 
- Created  new classes  called `Book`, and `Rental` ✔️ 
- Added relationship in between `Person`, `Book`, and `Rental` accroding to the project requirements ✅ 

***Thanks for visiting my pull request. If you have any questions feel free to ask 👍***

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Represents a book object with a title, an author, and associated rentals
+class Book
+  attr_accessor :title, :author
+
```
Comment (Shedrack-Sunday) [inline book.rb]: - [ ] Kindly inlude the Getters and Setters for `@rentals`.

## Sample code:
``attr_accessor :title, :author, :rentals``

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @raihan2bd    👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well so far in the project,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights   
- Good use of GitHub features ✔️   
- Nice code structure and use of classes and associations  👍 

  
# Required Changes  🌵 

- [x] You made some impressive implementations which I must say is a job well done. However, there are a few  features you missed

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (raihan2bd) [reply book.rb]: Hi @Shedrack-Sunday ✋,  Thanks a lot for your suggestion 👍 I will implement your suggestion before the next review. Thanks once again for reviewing my pull request.

Comment (Shedrack-Sunday) [reply book.rb]: You're welcome

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @raihan2bd , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**